modules/data-model/links/api/src/main/java/io/uhndata/cards/links/api/LinkUtils.java

```diff
@@ -56,6 +59,25 @@ public interface LinkUtils
      */
     Collection<Link> getLinks(Node source);
 
+    /**
+     * Retrieve all the links from a resource of a desired link type.
+     *
+     * @param source the node to get links from
+     * @param linkName The name of the link type, or a path starting from the root node to the link typs
+     * @return a collection of links, may be empty
+     */
+    Collection<Link> getLinksOfType(Node source, String linkName);
+
+    /**
+     * Search for an {@code cards:LinkDefinition} node with the desired name or path.
+     * @param session The session that should be used to find the link node
+     * @param linkName The name of the link node, or a path starting from the root node
+     * @return the specified Node for the desired link type
+     * @throws PathNotFoundException if the specified node does not exist
+     * @throws RepositoryException if another error occurs
+     */
+    Node getLinkType(Session session, String linkName) throws PathNotFoundException, RepositoryException;
```

Comment:
Shouldn't this return a `LinkDefinition` instead?

Reply:
This seems better to me:

```suggestion
    LinkDefinition getLinkDefinition(Session session, String type) throws PathNotFoundException, RepositoryException;
```

Reply:
Made private as the intended use case is better with Node and there's no need to expose this